.github/workflows/prcheck.yml

```
@@ -12,6 +16,15 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.ACTIONS_TOKEN }}
+
+      - name: Determine Workflow Branch
+        run: echo "TARGET_BRANCH=${{ github.event.pull_request.head.ref || github.ref_name }}" >> $GITHUB_ENV
```

actionlint

shellcheck reported issue in this script: SC2086:info:1:86: Double quote to prevent globbing and word splitting

(shellcheck)